ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/DefaultSchemaInjector.java

```diff
@@ -176,6 +185,25 @@ private SchemaAndId getSchema(
     return result.schemaAndId.get();
   }
 
+  private static void validateSchemaInference(
+      final FormatInfo formatInfo,
+      final boolean isKey
+  ) {
+    /*
+     * Do validation when schemaId presents or we need to infer. Conditions to meet:
+     *  1. If schema id is provided, format must support schema inference
```

Comment:
If this is the only validation that's being done, then the PR description is out of date, right?

Reply:
Yeah. Let me update that